crates/uplc/tests/pretty_tests.rs

```diff
@@ -109,7 +109,7 @@ fn constant_data_constr() {
     round_trip(
         Term::<Name>::Constant(
             Constant::Data(PlutusData::Constr(Constr::<PlutusData> {
-                tag: 1,
+                tag: 122,
```

Comment:
I am not 100% sure about this as pallas has a very weird way to represent Data's constr internally. `tag` suggests it should be indeed a CBOR tag, but I am also suspecting it is indeed expected to be the raw constr index (once parsed). 

The `any_constructor` is also confusing. I think I'll just fix it in Pallas directly at some point 🤔

Reply:
I fixed this before in naumachia. `tag` is the CBOR tag _unless_ the tag is too big to be stored as cbor value and is stored in `any_constructor` instead.

Yes. Please fix this in Pallas, this is the preferred way :)

Reply:
Okay interesting, appearently there I fixed things in the exact opposite direction. However, the implementation in the aiken state machine applied the transformation in constrData just as I did: https://github.com/aiken-lang/aiken/blob/7b452c21f01e5342b2c210e0a32f024c58ea2693/crates/uplc/src/machine/runtime.rs#L741

I am confused now as well.

Reply:
This line in Pallas clarifies that the implementation here is indeed correct. Pallas deserializes CBOR representations in exactly the way that I described above:

https://github.com/txpipe/pallas/blob/ca27aa91acf2bc579f3f1e4e8885cf2284fa90df/pallas-primitives/src/alonzo/model.rs#L1131

This actually implies that the naumachia fix is broken but this does not really matter here.